modules/login/php/endpoints/authentication.class.inc

```diff
@@ -0,0 +1,453 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used to retrieve info with authentication by the login form.
+ *
+ * PHP Version 7
+ *
+ * @category LORIS_Module
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\login\endpoints;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Middleware\ETagCalculator;
+use \LORIS\api\Endpoint;
+use SinglePointLogin;
+use State;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Login
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Authentication extends Endpoint implements ETagCalculator
```

Comment:
If this class is going to use endpoints, I think it makes sense to split them into different endpoints for password reset, account request, and normal logins. Doing a password reset is different from authenticating.

@xlecours maybe you could comment on this? I'm not fully confident about what makes the most sense for endpoints and the API.

Reply:
@johnsaigle I'm going to do similar to this PR from the comments received from @driusan about the issue_tracker PR.